OneCommunityGlobal/HGNRest
# Nathan permissions merge

# Description
Fixes #6 (HIGH) Verify that permissions are implemented on the back end and not just the front end.
See front-end PR #

## Related PRS (if any):
This backend PR is related to the #931 frontend PR.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/931

## Main changes explained:
- Updated createInitialPermissions to combine front-end and back-end permissions into a single permission set.
- Update rolesController.js, taskController.js, teamController.js, and userProfileController.js to have permissions set up. 
- Updated role.js to use different db collection (to not delete data for current system while still testing. can clean up old permissions in a later PR.)
- Updated permissions.js to only check one set of permissions.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. The front-end should work the same.

## Screenshots or videos of changes:
Everything should look the same other than the code.
